glue/flagging_script_glue/flagging.py

```diff
@@ -1257,7 +1279,14 @@ def create_name_match(row: pd.Series) -> str:
     Outputs:
         value (str or None): string match if applicable, None otherwise
     """
-    if row["sv_buyer_id"] == row["sv_seller_id"] and row["sv_buyer_id"] != "Empty Name":
+    if (
+        row["sv_buyer_id"] == row["sv_seller_id"]
+        and row["sv_buyer_id"] != "Empty Name"
+        # Prevents the same legal entity as counting as a family name match
+        and row["sv_transaction_type"] != "legal_entity-legal_entity"
+        # Boots out matches on a single last initial
+        and len(row["sv_buyer_id"]) > 1
+    ):
```

Comment:
There are two fixes here:

- First, if a legal entity was both a buyer and a seller with the exact string name, it would be classified as a `Family sale`. This line of code fixes that: `and row["sv_transaction_type"] != "legal_entity-legal_entity"`
- Second, there were multiple instances where the last token picked up by the regex was a single initial which resulted in a family sale name match. We fix that here: `and len(row["sv_buyer_id"]) > 1`

Reply:
**praise:** Nice! Good catches.